### ISSUE TITLE
Titanic - Pclass feature is preprocessed as a categorical feature.

### ISSUE THREAD

**Issue by Phylake1337** [ageron/handson-ml2#90]
Pclass is a numerical feature which its values could be 1,2 or 3 that describe ticket class.
So it should be preprocessed as a numerical feature, please correct me if I am wrong.
`
cat_pipeline = Pipeline([
        ("select_cat", DataFrameSelector(["Pclass", "Sex", "Embarked"])),
        ("imputer", MostFrequentImputer()),
        ("cat_encoder", OneHotEncoder(sparse=False)),
    ])
`
Hence, Pclass should be removed from this cell, and be written in num_pipline cell.
`num_pipeline = Pipeline([
        ("select_numeric", DataFrameSelector(["Age", "SibSp", "Parch", "Fare"])),
        ("imputer", SimpleImputer(strategy="median")),
    ])
`
Any thougths?

**Comment by VirajVaitha123** [comment]
OK so while I don't 

When we convert it using one hot encoding it is indeed a numerical feature.

So I geuss the question is why One Hot Encode vs Ordinal Encoding?

Having labels of this nature (1,2,3,4,5) is often used when the size of the number makes a difference. 

For example if you have exam results of A, B, C, D, E(5, 4,3,2,1) we are choosing to say A is closer to B than E.

Whereas if we had cat, dog, fox in a column, we can only have one of these classes at once. OneHotEncoding makes more sense. OneHotEncoding would result in 3 columns where you could have 1,0,0(cat) or 0,1,0 dog or 0,1,0(fox). If we encoded as 1,2,3, then we are implying say the dog is more similar to the fox than the cat and this would be incorrect.

Finally, with this example here you could argue Pclass can be used as a numerical column immediately based on the knowledge I have provided. 

We should try the model with both types of encodings and compare the evaluation results. 

It would be interesting to see what other people think. I am still learning and not entirely confident on this area too.  So let's see what other people think.

Hope this helps, 

Thanks

**Comment by Phylake1337** [comment]
Thanks for your help, your explanation makes sense as I read the same thing before, but practically in this kind of problem, people on the shape with 1st class ticket are more likely to survive.
So in my opinion which is based on your thought, people with 2nd class are more likely to survive than those with the 3rd class ticket class. hence, people with 2nd degree are more similar to those with 1st class ticket than 3rd class people.
However, I implemented the solution that considers pclass as a numerical feature(1 column that contains (1,2,3)),and the result was the same.

regards,

**Comment by VirajVaitha123** [comment]
Nice one :) 

I'm glad you tested both was curious about this.

I agree with you seems like that data made more sense as nominal. Alternatively I wonder if forcing 2nd class = 5 and third class = 20 would make a difference.

So 1, 5 and 20. Sounds a little stupid and may introduce bias now reading it back lol

**Comment by ageron** [comment]
Thanks for your question @Phylake1337, and sorry for the late response. Big thank you to @VirajVaitha123 for your excellent answer, I don't have much to add.
In my model I chose not to make any assumption about the relationships between the classes, hence the one-hot encoding. If I was certain that the survival order was 1, 2, 3 and not 1, 3, 2, then indeed encoding the class directly as a number would make perfect sense.

By the way, consider the passenger's age. Imagine if children and elderly people were given priority for the life boats. If so, the model might get a better performance if we bucketized the age feature into, say 10 age groups, and then encoded each group using one-hot-encoding. For example, a 12-year old child might be in age group 2, and this would be encoded as [0, 0, 1, 0, 0, 0, 0, 0, 0, 0].  This would make it easier for the model to treat age groups differently. Otherwise, it would need to learn the non-linear relationship between age and survival, which is typically harder for most models.

In short, the prior knowledge you have about the topic, and the assumptions you make determine how you build your model, and ultimately affects its performance.

Hope this helps.

I'll close this issue, but please feel free to reopen it if it's still unclear.
